app/scripts/lib/eth-store.js

```diff
@@ -7,140 +7,122 @@
-}
-
-//
-// public
-//
 
-EthereumStore.prototype.getState = function () {
-  const self = this
-  return clone(self._currentState)
-}
-
-EthereumStore.prototype.addAccount = function (address) {
-  const self = this
-  self._currentState.accounts[address] = {}
-  self._didUpdate()
-  if (!self.currentBlockNumber) return
-  self._updateAccount(address, () => {
-    self._didUpdate()
-  })
-}
+  //
+  // public
+  //
 
-EthereumStore.prototype.removeAccount = function (address) {
-  const self = this
-  delete self._currentState.accounts[address]
-  self._didUpdate()
-}
+  addAccount (address) {
+    const accounts = this.getState().accounts
+    accounts[address] = {}
+    this.updateState({ accounts })
+    if (!this._currentBlockNumber) return
+    this._updateAccount(address)
+  }
 
-EthereumStore.prototype.addTransaction = function (txHash) {
-  const self = this
-  self._currentState.transactions[txHash] = {}
-  self._didUpdate()
-  if (!self.currentBlockNumber) return
-  self._updateTransaction(self.currentBlockNumber, txHash, noop)
-}
+  removeAccount (address) {
+    const accounts = this.getState().accounts
+    delete accounts[address]
+    this.updateState({ accounts })
+  }
 
-EthereumStore.prototype.removeTransaction = function (address) {
-  const self = this
-  delete self._currentState.transactions[address]
-  self._didUpdate()
-}
+  addTransaction (txHash) {
+    const transactions = this.getState().transactions
+    transactions[txHash] = {}
+    this.updateState({ transactions })
+    if (!this._currentBlockNumber) return
+    this._updateTransaction(this._currentBlockNumber, txHash, noop)
+  }
 
+  removeTransaction (txHash) {
+    const transactions = this.getState().transactions
+    delete transactions[txHash]
+    this.updateState({ transactions })
+  }
 
-//
-// private
-//
 
-EthereumStore.prototype._didUpdate = function () {
-  const self = this
-  var state = self.getState()
-  self.emit('update', state)
-}
+  //
+  // private
+  //
+
+  _updateForBlock (block) {
+    const blockNumber = '0x' + block.number.toString('hex')
+    this._currentBlockNumber = blockNumber
+    async.parallel([
+      this._updateAccounts.bind(this),
+      this._updateTransactions.bind(this, blockNumber),
+    ], (err) => {
+      if (err) return console.error(err)
+      this.emit('block', this.getState())
+    })
+  }
 
-EthereumStore.prototype._updateForBlock = function (block) {
-  const self = this
-  var blockNumber = '0x' + block.number.toString('hex')
-  self.currentBlockNumber = blockNumber
-  async.parallel([
-    self._updateAccounts.bind(self),
-    self._updateTransactions.bind(self, blockNumber),
-  ], function (err) {
-    if (err) return console.error(err)
-    self.emit('block', self.getState())
-    self._didUpdate()
-  })
-}
+  _updateAccounts (cb = noop) {
+    const accounts = this.getState().accounts
+    const addresses = Object.keys(accounts)
+    async.each(addresses, this._updateAccount.bind(this), cb)
```

Comment:
Oh wow, our `eth-store` getting an overhaul?

Have you considered promisifying this?